server/query.go

```diff
@@ -0,0 +1,127 @@
+package server
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tm "github.com/tendermint/tendermint/proto/tendermint/types"
+	coretypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+// QueryBlocksByEvents performs a search for blocks for a given set of events
+// via the Tendermint RPC. An event takes the form of:
+// "{eventAttribute}.{attributeKey} = '{attributeValue}'". Each event is
+// concatenated with an 'AND' operand. It returns a slice of Info object
+// containing blocks and metadata. An error is returned if the query fails.
+// If an empty string is provided it will order blocks by asc
+func QueryBlocksByEvents(clientCtx client.Context, events []string, page, limit int, orderBy string) (*sdk.SearchBlocksResult, error) {
+	if len(events) == 0 {
+		return nil, errors.New("must declare at least one event to search")
+	}
+
+	if page <= 0 {
+		return nil, errors.New("page must be greater than 0")
+	}
+
+	if limit <= 0 {
+		return nil, errors.New("limit must be greater than 0")
+	}
+
+	// XXX: implement ANY
```

Comment:
Let's not parse or do anything special with the query given. i.e. pass it directly to Tendermint.

Ref: https://github.com/cosmos/cosmos-sdk/pull/14758/